src/main/java/com/nekomaster1000/infernalexp/world/gen/structures/BastionOutpostStructure.java

```diff
@@ -35,32 +43,114 @@ public BastionOutpostStructure(Codec<NoFeatureConfig> codec) {
 
 	@Override
 	public StructureSeparationSettings getSeparationSettings() {
-		return new StructureSeparationSettings(4, 2, 720435943);
+		return new StructureSeparationSettings(2, 1, 720435943);
 	}
 
-//	@Override
-//	protected boolean func_230363_a_(ChunkGenerator chunkGenerator, BiomeProvider biomeProvider, long seed, SharedSeedRandom chunkRandom, int chunkX, int chunkZ, Biome biome, ChunkPos chunkPos, NoFeatureConfig config) {
-//		return super.func_230363_a_(chunkGenerator, biomeProvider, seed, chunkRandom, chunkX, chunkZ, biome, chunkPos, config);
-//	}
-
 	@Override
 	public boolean shouldTransformLand() {
 		return true;
 	}
 
+    @Override
+    protected boolean func_230363_a_(ChunkGenerator chunkGenerator, BiomeProvider biomeProvider, long seed, SharedSeedRandom chunkRandom, int chunkX, int chunkZ, Biome biome, ChunkPos chunkPos, NoFeatureConfig config) {
+        SharedSeedRandom random = new SharedSeedRandom(seed + (chunkX * (chunkZ * 17)));
+
+        // Makes cheap check first, if it passes this check, it does a more in-depth check
+        if (super.func_230363_a_(chunkGenerator, biomeProvider, seed, chunkRandom, chunkX, chunkZ, biome, chunkPos, config)) {
+
+            int posX = chunkX << 4;
+            int posZ = chunkZ << 4;
+            int posY = getYPos(chunkGenerator, posX, posZ, random);
+
+            // Checks 9 points within about a chunk of the initial location
+            for (int curX = posX - 8; curX <= posX + 8; curX += 8) {
+                for (int curZ = posZ - 8; curZ <= posZ + 8; curZ += 8) {
+
+                    // Starts 5 blocks below to check for solid land in each column
+                    BlockPos.Mutable mutable = new BlockPos.Mutable();
+                    mutable.setPos(curX, posY - 5, curZ);
```

Comment:
```suggestion
                    BlockPos.Mutable mutable = new BlockPos.Mutable(curX, posY - 5, curZ);
```

Reply:
The thing about moving func_240363_a_ to a super class is that each structure type uses different values for the checks, so I'd still just have to override it in each subclass specific structure anyway. I could move the getYPos method, but that's about the only one that makes sense.

Reply:
Couldn't you just add new arguments for those values?

Reply:
Fair enough, but there's a lot of them. There's how far away to go from the initial point, how much to increment after each sample, how vertically to check for land, and how vertically to check for air above the structure. That feels like a bit much but I can do it if you think I should.

Reply:
Hmm... if it's that much, then don't implement it in the superclass. (also this thread is in the wrong place, this is about a suggestion for the code on line 70 LMAO)